python/pyspark/mllib/linalg.py

```diff
@@ -670,6 +668,23 @@ def toArray(self):
         """
         return self.values.reshape((self.numRows, self.numCols), order='F')
 
+    def toSparse(self):
+        """Convert to Sparse Format"""
+        numRows = self.numRows
```

Comment:
This could be simplified:

~~~python
indices = np.nonzero(self.values)[0]
colPtrs = np.cumsum(np.bincount(indices / self.numRows + 1))
values = self.values[indices]
rowIndices = indices % self.numRows
return SparseMatrix(self.numRows, self.numCols, colPtrs, indices, values)
~~~